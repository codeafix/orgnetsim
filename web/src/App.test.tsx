import React from 'react';
import {render, waitFor, screen} from '@testing-library/react';
import '../node_modules/bootstrap/dist/css/bootstrap.css';
import './index.css';
import {BrowserRouter} from 'react-router-dom';
import App from './App';
import { expect, test, vi } from 'vitest'
import { act } from 'react-dom/test-utils';

vi.mock('./API/api');

test('renders without crashing', async () => {
  var result:any;
  await act(async () => {
    result = render(
      <BrowserRouter>
          <App />
      </BrowserRouter>
    );
  });
  expect(result).toBeDefined();
  expect(result.asFragment()).toMatchSnapshot();
});

