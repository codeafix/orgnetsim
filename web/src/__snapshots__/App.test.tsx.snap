// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <header />
    <main>
      <div
        class="container-fluid"
      >
        <h1>
          <img
            alt="logo"
            src="/src/logo.svg"
            style="height: 60px; width: 60px;"
          />
          Simulation Set
        </h1>
        <p>
          Some notes
        </p>
        <h2>
          List of Simulations
          <button
            class="btn btn-primary float-right btn btn-primary"
            type="button"
          >
            Add
          </button>
        </h2>
        <div
          class="card-deck"
        >
          <div
            class="card border-info"
            style="width: 18rem;"
          >
            <div
              class="card-header"
            >
              <div
                class="card-title h5"
              >
                test1
                <svg
                  class="float-right"
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5m2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5m3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0z"
                  />
                  <path
                    d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4zM2.5 3h11V2h-11z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="card-body"
            >
              <p
                class="card-text"
              >
                test1desc
              </p>
            </div>
            <div
              class="card-footer"
            >
              <a
                class="btn btn-primary"
                href="/simulation/1"
                role="button"
              >
                Open
              </a>
            </div>
          </div>
          <div
            class="card border-info"
            style="width: 18rem;"
          >
            <div
              class="card-header"
            >
              <div
                class="card-title h5"
              >
                test2
                <svg
                  class="float-right"
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5m2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5m3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0z"
                  />
                  <path
                    d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4zM2.5 3h11V2h-11z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="card-body"
            >
              <p
                class="card-text"
              >
                test2desc
              </p>
            </div>
            <div
              class="card-footer"
            >
              <a
                class="btn btn-primary"
                href="/simulation/2"
                role="button"
              >
                Open
              </a>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
